### PR TITLE
[Train] Remove `CheckpointConfig` from `ray.train`

### PR DESCRIPTION
**Pull request by amogkam**
Signed-off-by: Amog Kamsetty <amogkamsetty@yahoo.com>

Quick follow up to https://github.com/ray-project/ray/pull/27005 to add back `CheckpointConfig` to `ray.train`

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/__init__.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
     world_size,
 )
 from ray.train.trainer import Trainer, TrainingIterator
+from ray.air.config import CheckpointConfig
```
Comment (matthewdeng) [inline python/ray/train/__init__.py:16]: Do we want to expose this alias? Or just have users directly use the one from `air`?

Comment (amogkam) [reply python/ray/train/__init__.py:16]: This PR is just to match previous behavior which got removed in #27005 (the current Ray Train docs show `from ray.train import CheckpointConfig`).

We can remove the alias in a follow up.

Comment (matthewdeng) [reply python/ray/train/__init__.py:16]: Okay got it should we add a deprecation warning [here](https://github.com/amogkam/ray/blob/5febe7488b8f014a5d0b5399a2b8b9f4f09e7dd3/python/ray/train/checkpoint.py) for this?

**Review by matthewdeng — APPROVED** [review]

File: python/ray/train/__init__.py (hunk 1 above)
Comment (matthewdeng) [reply python/ray/train/__init__.py:16]: Ah  that one is for `train.CheckpointStrategy`, not `train.CheckpointConfig`

**Comment by amogkam** [comment]
Oh nvm, you are right, the most recent docs all show CheckpointConfig from ray.air. Going to close this.

**Comment by matthewdeng** [comment]
@amogkam  should we get rid of `"CheckpointConfig",` from `__all__` then ?

**Review by krfricke — APPROVED** [review]

**Comment by amogkam** [comment]
yeah just updated

**Pull request merged**